src/plugins/mongodb-core.js

```diff
@@ -129,6 +130,10 @@ function isObject (val) {
   return typeof val === 'object' && val !== null && !(val instanceof Array)
 }
 
+function isBSON (val) {
+  return val && val._bsontype && typeof val.toJSON === 'function'
```

Comment:
The reasoning was that you need to check for `_bsontype` to validate it's BSON, and I added a check for `toJSON` just as a safety net so that if this method doesn't exist for any reason it will not crash.

Reply:
Makes sense.